x-pack/plugins/security_solution/public/management/components/endpoint_responder/lib/console_commands_definition.ts

```diff
@@ -523,6 +524,87 @@ export const getEndpointConsoleCommands = ({
       privileges: endpointPrivileges,
     }),
   });
+  if (crowdstrikeRunScriptEnabled) {
+    consoleCommands.push({
+      name: 'runscript',
+      about: getCommandAboutInfo({
+        aboutInfo: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.about,
+        isSupported: doesEndpointSupportCommand('runscript'),
+      }),
+      RenderComponent: () => null,
+      meta: {
+        agentType,
+        endpointId: endpointAgentId,
+        capabilities: endpointCapabilities,
+        privileges: endpointPrivileges,
+      },
+      exampleUsage: `runscript -Raw=\`\`\`Get-ChildItem .\`\`\` -CommandLine=""`,
+      helpUsage: `
+Command Examples for Running Scripts:
+
+1. Executes a script saved in the CrowdStrike cloud with the specified command-line arguments.
+
+   runscript -CloudFile="CloudScript1.ps1" -CommandLine="-Verbose true"
+
+2. Executes a script saved in the CrowdStrike cloud with the specified command-line arguments and a 180-second timeout.
+
+   runscript -CloudFile="CloudScript1.ps1" -CommandLine="-Verbose true" -Timeout=180
+
+3. Executes a raw script provided entirely within the "-Raw" flag.
+
+   runscript -Raw="Get-ChildItem."
+
+4. Executes a script located on the remote host at the specified path with the provided command-line arguments.
+
+   runscript -HostPath="C:\\temp\\LocalScript.ps1" -CommandLine="-Verbose true"
+
+`,
+      exampleInstruction: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.about,
+      validate: capabilitiesAndPrivilegesValidator(agentType),
+      mustHaveArgs: true,
+      args: {
+        Raw: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.raw.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        CloudFile: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.cloudFile.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        CommandLine: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.commandLine.about,
+          mustHaveValue: 'non-empty-string',
+        },
+        HostPath: {
+          required: false,
+          allowMultiples: false,
+          about: CROWDSTRIKE_CONSOLE_COMMANDS.runscript.args.hostPath.about,
+          mustHaveValue: 'non-empty-string',
+        },
```

Comment:
This is a bit tricky, since CrowdStrike lets you add multiple of these arguments, and then the strange behavior I explained during the meeting happened. But I agree with you that our approach would be to use `exclusiveOr` and just let them use one at a time. 
Added `exclusiveOr` 👍 Thank you, I also added a description to exclusiveOr type.